python/lsst/ts/wep/task/calcZernikesNeuralTask.py

```diff
@@ -0,0 +1,1964 @@
+
+        self.log.info("Set DonutStamps metadata: DET_NAME='%s', CAM_NAME='%s', DFC_TYPE='%s'",
+                     detectorName, cameraName, defocalType)
+
+        # Debug: Check what metadata keys are available after setting
+        self.log.debug("Metadata keys after setting: %s", list(donutStampsObj.metadata.names()))
+        self.log.debug("DET_NAME value: %s (type: %s)", donutStampsObj.metadata.get("DET_NAME", "NOT_FOUND"),
+                      type(donutStampsObj.metadata.get("DET_NAME", "NOT_FOUND")))
+        # Store the detector name for potential recovery later
+        self._detector_name = detectorName
+        self.log.debug("Stored detector name for recovery: '%s'", self._detector_name)
+
+        # Override _refresh_metadata to preserve our scalar DET_NAME
+        original_refresh = donutStampsObj._refresh_metadata
+        def preserve_det_name_refresh() -> None:
+            original_refresh()
+            # Restore our scalar DET_NAME if it was overwritten by an array
+            det_name_value = donutStampsObj.metadata.get("DET_NAME", None)
+            if isinstance(det_name_value, list):
+                self.log.debug("Restoring scalar DET_NAME after _refresh_metadata() (was array)")
+                donutStampsObj.metadata["DET_NAME"] = self._detector_name
+            elif det_name_value is None or det_name_value == "Unknown":
+                self.log.debug("Restoring scalar DET_NAME after _refresh_metadata() (was missing/unknown)")
+                donutStampsObj.metadata["DET_NAME"] = self._detector_name
+        donutStampsObj._refresh_metadata = preserve_det_name_refresh
+
+        # Add exposure metadata using safe extraction
+        exp_metadata = self._get_exposure_metadata(exposure)
+
+        # Set metadata with safe fallback values
+        donutStampsObj.metadata["VISIT"] = exp_metadata.get('visit_id', -1)
+        donutStampsObj.metadata["BORESIGHT_ROT_ANGLE_RAD"] = exp_metadata.get(
+            'boresight_rot_angle_rad', 0.0
+        )
+        donutStampsObj.metadata["BORESIGHT_PAR_ANGLE_RAD"] = exp_metadata.get(
+            'boresight_par_angle_rad', 0.0
+        )
+        donutStampsObj.metadata["BORESIGHT_ALT_RAD"] = exp_metadata.get(
+            'boresight_alt_rad', float('nan')
+        )
+        donutStampsObj.metadata["BORESIGHT_AZ_RAD"] = exp_metadata.get(
+            'boresight_az_rad', float('nan')
+        )
+        donutStampsObj.metadata["BORESIGHT_RA_RAD"] = exp_metadata.get(
+            'boresight_ra_rad', float('nan')
+        )
+        donutStampsObj.metadata["BORESIGHT_DEC_RAD"] = exp_metadata.get(
+            'boresight_dec_rad', float('nan')
+        )
+        donutStampsObj.metadata["MJD"] = exp_metadata.get('mjd', float('nan'))
+
+        self.log.info("Constructed %d DonutStamps with WCS and metadata", len(donutStamps))
+        return donutStampsObj
+
+    def donutStampsToQTable(self, donutStamps: DonutStamps) -> QTable:
```

Comment:
As far as I can tell you don't use this function anywhere in this PR, why are you creating it?